datamining/procedures/filter_members.py

```diff
@@ -3,23 +3,20 @@
 from members.models import Member
 
 
-def filter_members(found_committees, found_privacy_settings, before, after, include_honorary, only_blacklisted, dms):
-    query = None
+def filter_members(found_committees, found_privacy_settings, on, include_honorary, only_blacklisted, dms):
     # membership period checks
-    after_query = Q(membershipperiod__start_date__lte=after)
-    before_query = Q(membershipperiod__end_date__gte=before)
+    after_query = Q(membershipperiod__start_date__lte=on)
+    before_query = Q(membershipperiod__end_date__gte=on)
     before_query_honorary = Q(membershipperiod__end_date__isnull=True)
-    if after:
-        query = after_query
-    if before:
-        combined_before_query_honorary = before_query
-        if include_honorary:
-            combined_before_query_honorary = (before_query | before_query_honorary)
-
-        if query is None:
-            query = combined_before_query_honorary
-        else:
-            query = query & combined_before_query_honorary
+    query = after_query
```

Comment:
The variable names are quite confusing here.. They already were before, but this may be the right time to fix them :).

I think part of the query could be simplified to something like:
membership_period_filter = before_part & after_part
if include honorary:
  membership_period_filter = before_part & (after_part | honorary_filter

Then it ends on a simple:
query= membership_period_filter
if settingX:
 query = query & X
etc